src/VisualStudio/Core/Impl/CodeModel/InternalElements/AbstractCodeType.cs

```diff
@@ -165,19 +165,34 @@ public override bool IsCodeType
 
         public void RemoveMember(object element)
         {
-            var codeElement = ComAggregate.TryGetManagedObject<AbstractCodeElement>(element);
+            // Is this an EnvDTE.CodeElement that we created? If so, try to get the underlying code element object.
+            var abstractCodeElement = ComAggregate.TryGetManagedObject<AbstractCodeElement>(element);
 
-            if (codeElement == null)
+            if (abstractCodeElement == null)
             {
-                codeElement = ComAggregate.TryGetManagedObject<AbstractCodeElement>(this.Members.Item(element));
+                var codeElement = element as EnvDTE.CodeElement;
+                if (codeElement != null)
+                {
+                    // Is at least an EnvDTE.CodeElement? If so, try to retrieve it from the Members collection by name.
+                    // Note: This might throw an ArgumentException if the name isn't found in the collection.
+
+                    abstractCodeElement = ComAggregate.TryGetManagedObject<AbstractCodeElement>(this.Members.Item(codeElement.Name));
+                }
+                else if (element is string || element is int)
```

Comment:
I don't think `element is string` could ever return true because the previous `if (codeElement != null)` should cover it.